Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs

```diff
@@ -343,6 +360,13 @@ private void OnEnd(object exception, object thisObj, object returnValue)
                             // ObjectDisposedException is expected here in the following sequence: httpWebRequest.GetResponse().Dispose() -> httpWebRequest.GetResponse()
                             // on the second call to GetResponse() we cannot determine the statusCode.
                         }
+
+                        var targetIkeyHash = responseObj.Headers[ComponentCorrelationConstants.TargetInstrumentationKeyHeader];
```

Comment:
Are you specifically talking about the line var targetIkeyHash = responseObj.Headers[ComponentCorrelationConstants.TargetInstrumentationKeyHeader]; 
Why do you think it will ever throw? Or are you talking about the catch block above where we are swallowing ObjectDisposeException?